ledger/acctupdates.go

```diff
@@ -246,66 +255,36 @@ func (au *accountUpdates) waitAccountsWriting() {
 	au.accountsWriting.Wait()
 }
 
+// close closes the accountUpdates, waiting for all the child go-routine to complete
 func (au *accountUpdates) close() {
 	if au.ctxCancel != nil {
 		au.ctxCancel()
 	}
 	au.waitAccountsWriting()
+	// this would block until the commitSyncerClosed channel get closed.
+	<-au.commitSyncerClosed
 }
 
-func (au *accountUpdates) lookup(rnd basics.Round, addr basics.Address, withRewards bool) (data basics.AccountData, err error) {
+// Lookup returns the accound data for a given address at a given round. The withRewards indicates whether the
+// rewards should be added to the AccountData before returning. Note that the function doesn't update the account with the rewards,
+// even while it could return the AccoutData which represent the "rewarded" account data.
```

Comment:
> Note that the function doesn't update the account with the rewards, 

It is confusing here. Will the returned information be updated? And consequently differ from what is in the account?

Reply:
This function does not update the balance when retrieving a account data with rewards. It only does the calculation.